### ISSUE TITLE
[508 COGNITION, SCREENREADER] `va-text-input` maxlength notice

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#38775]
## User Story or Problem Statement

When the `va-text-input` web component includes a `maxlength` property, the user is not aware that there is a limitation on the input until the maximum character limit is reached. When reached, a `small` element is added below the input with an `aria-live="polite"` attribute. This was not announced while using JAWS in Edge on a Windows system.

## Details

There are several problems with this:

- The screenreader user will not know that there is a limitation prior to typing
- Not all screen readers will only announce `aria-live` regions that don't already exist on the page, i.e. It may not work on dynamically added elements.
- The `(Max. {#} characters)` message disappears after the last character is removed

## Acceptance Criteria

- [ ] Character limit should be announced and known before the user starts typing in the input
- [ ] Input should be able to accomodate a label, hint text, max character message and any error messages

## Environment

* Operating System, including Windows 10
* Browser, including Edge
* Screenreading device, if applicable: JAWS
* Server destination: Design Storybook

## Steps to Recreate

1. Go to https://design.va.gov/storybook/?path=/docs/components-va-text-input--max-length
2. Start screen reading device 
3. Focus on the Max Length input
4. Enter 16 characters into the input
5. The "Max. 16 characters" message will appear below the input
6. The combination of screen reader & browser may or may not announce this message
7. Remove one character and the max message disappears

## Proposed Solution (if known)

I met with @CourtneyChambers77 and she said including the max character message above the input would be ideal. Incorporate it into the label (as in this [hint text guidance](https://design.va.gov/experimental-design/hint-text#inline-with-the-label)), or below any actual hint text.

cc: @sshein 

## Screenshots or Trace Logs

Screenshot from design.va.gov storybook max length example:
<img width="505" alt="Input with numbers 1-9 and 0-6 with a max 16 characters in parentheses below the input" src="https://user-images.githubusercontent.com/136959/159083727-97ca5a7d-2c5a-42fe-b078-d270f0380bdc.png">

**Comment by SarahKay8** [comment]
Confirmed; User is not aware that there is a limitation on the input until the maximum character limit is reached 

Solutions: 
When the limit is reached we can detect that with JS and then announce it to assistive technologies e.g. screen-readers by insert the following after the <input/>

<p class="visually-hidden" aria-live="assertive" role="alert">Limit reached. You can only use 16 characters in this field.</p>

Note: remember to remove it if/when within the limit again.

we can also  hide .visually-hidden off-screen with CSS.. or this information could actually be useful for everyone so you could omit the .visually-hidden class if it fits your design.

And to let screen-reader users know about the limit you could put <span class=visually-hidden>Max 16 characters</span> inside your <label> element. (and same goes here, perhaps all users could benefit from having that info visible..)

**Comment by SarahKay8** [comment]
Update has been added via storybook. Testing of chromatic link will begin on ticket #580 -Add maxlength hint text to va-text-input
